plugins/inputs/socket_listener/socket_listener.go

```diff
@@ -52,18 +54,27 @@ func (sl *SocketListener) SetParser(parser telegraf.Parser) {
 
 func (sl *SocketListener) Start(acc telegraf.Accumulator) error {
 	// Create the callbacks for parsing the data and recording issues
-	onData := func(_ net.Addr, data []byte) {
+	onData := func(_ net.Addr, data []byte, receiveTime time.Time) {
 		metrics, err := sl.parser.Parse(data)
+
 		if err != nil {
 			acc.AddError(err)
 			return
 		}
+
 		if len(metrics) == 0 {
 			once.Do(func() {
 				sl.Log.Debug(internal.NoMetricsCreatedMsg)
 			})
 		}
+
 		for _, m := range metrics {
+			switch sl.TimeSource {
+			case "", "metric":
+			case "receive_time":
+				m.SetTime(receiveTime)
+			}
+
```

Comment:
I think this should not be here... :-)

Reply:
It should, since service plugins do not work with #15917 

https://github.com/influxdata/telegraf/pull/15891#issuecomment-2361484569

Reply:
Yeah but please make this an own PR to separate feature introduction!